# scRNA-Population-Analysis
The purpose of this analysis is to determine what populations are present, specifically in a microenvironment for scRNA-seq data.
The code is based around Seurat with addtional computational analysis for deeper insights. File used for tertiary scRNA analysis is a filtered feature_bc_matrix file ('bc' stands for barcode). 
This file is a sparse counts matrix file of gene expression data of a subset of barcodes identified as corresponding to actual cells.
This is the file that stores gene expression counts based on UMIs (unique molecular identifiers).
```r
# Call in necessary libraries that provide pre-written, reusable functions and tools to do job.
library(dplyr)
library(Seurat)
library(patchwork)
library(readr)
```
Ensure you set your working directory to where you have the "filtered_feature_bc_matrix" files stored.

```r
# Set working directory to where your files are. Be sure to include " " in the code below. 'data' now becomes that pathway. Do this for each matrix file (data, data1, data2...)
setwd("/home/user/documents")
data <- "/home/user/documents/sampleID/filtered_feature_bc_matrix"
```

The filtered_feature_bc_matrix is typically a directory containing files like barcodes.tsv.gz, features.tsv.gz, and matrix.mtx.gz, generated by 10X Genomics' Cell Ranger pipeline.
Read this pathway in.  I usually name the new directory as the sample ID.
```r
# Example
CLL1234 <- Read10X(data)
CLL5678 <- Read10X(data1)
```
Now you want to create the Seurat object from this count matrix.  A "Seurat object" is designed for scRNA-seq analysis that stores multiple types of data.
Types of data stored in this object: expression matrix (genes vs. cells), metadata(e.g., cell annotations), analysis results (e.g., clustering, dimensionality reduction).





